src/app/services/user/UserDAO.ts

```diff
@@ -56,9 +60,39 @@ export function get(name : String, email? : String) : Promise <User> {
 	return wrapFindOne<UserDocument, User>(UserModel.findOne(query).select('-password'));
 }
 
+/*
+ * Fetching user by email. password is excluded.
+ */
+export function getByEmail(email : String) : Promise <User> {
+	emptyCheck(email);
+	const query : any = { email: email };
+
+	return wrapFindOne<UserDocument, User>(UserModel.findOne(query).select('-password'));
+}
+
+/*
+ * Fetching user by ID parameter. Password excluded.
+ */
+export function getByID(id : String) : Promise <User> {
+	return UserModel.findOne({'_id': id})
+	.select('-password').exec()
+	.then(res => (res === null)
+				? Promise.reject(new NotFoundError('User not found'))
+				: Promise.resolve(res)
+	);
+}
+
 export function save(user : User) : Promise <User> {
 	emptyCheck(user);
-	return wrapSave<User>(new UserModel(user).save());
+
+	if (user.password !== undefined) {
+		return bcrypt.hash(user.password, config.get('auth.bcrypt.rounds')).then((hash) => {
```

Comment:
I have seen this being reworked in another branch (if I wasn't dreaming that) to an external update-password function, like requested. If not, it still has to be fixed. What is the general status of this task?